esp/platform/txsummary.cpp

```diff
@@ -34,6 +35,7 @@ bool operator < (const StringAttr& a, const StringAttr& b)
 CTxSummary::CTxSummary(unsigned creationTime)
 : m_creationTime(creationTime ? creationTime : msTick())
 {
+    getESPContainer()->getEspMetrics()->pCountRequests->inc(1);
```

Comment:
As we've discussed offline, this code could be running outside of ESP in which case getESPContainer() will return nullptr.  You said you wanted to avoid an if check each time.  If that's the case you can wrap the metrics object globally and have it stubbed out when getESPContainer() is null.  Then this call would be a no-op for that scenario.